design-doc.md

```diff
@@ -0,0 +1,137 @@
+# Design Document: Adding Support for Next Link Paging with Re-injected Query Parameters
+
+## Overview
+
+Currently, the client treats the `nextLink` in paging as opaque and does not inject query parameters that were part of the initial request when following the next link. This proposal aims to enhance the client by adding capability to reinject query parameters into next requests for paging operations.
+
+## Problem
+
+This is an existing issue in 1-2 brownfield Azure services: Keyvault and potentially AppConfig.
+
+### Keyvault
+
+KV only has one instance in `listCertificates`, there is an optional query parameter `includePending`, which dictates whether to include pending certificates in the returned list of certificates. The service does not automatically reinject the value of `includePending` into the next link, so for all subsequent calls during paging, you will have to reinject `includePending=True` if you would like to continue seeing pending certificates in subsequent pages.
+
+### AppConfig
+
+I can't get in touch with anyone on the client-side team working with AppConfig since they're all no longer with us.
+
+The examples used throughout the rest of this doc will reference the `includePending` issue from KV.
+
+## Requirements
+
+1. Solves the issue for our 3 Azure services: KeyVault, Storage, AppConfig.
+2. Open question of whether this support is required for 3p. For Azure services, requiring clients to reinject parameters is against guidelines. However, should we restrict 3p authors from describing this in their specs, even if it might not be best practice?
+3. Work with templates
+
+## Design Proposals
+
+### Fixing `@nextPageOperation` in typespec-azure
+
+```tsp
+@get nextPage is Azure.Core.Foundations.Operation<{ nextLink: string, includePending?: string }, Azure.Core.Page<Certificate>>;
+
+@nextPageOperation(
+    nextPage,
+    { nextLink: ResponseProperty<"nextLink">, includePending: RequestParameter<"includePending"> }
+)
+list is Azure.Core.ResourceList<Certificate>;
+```
+
+#### Pros
+
+1. Definitely easiest path forward, since all of the code is pretty much here for us. All we have to do is fix [this](https://github.com/Azure/typespec-azure/issues/1880) issue
+2. No new decorators or syntax introduced
+
+#### Cons
+
+1. Ugly definition
+2. We are gradually moving away from paging from `azure-core`, and unclear if there would be a counterpart in unbranded paging, so this won't be a good idea for 3p support or non-brownfield service support
```

Comment:
Are you implying that `nextPageOperation` doesn't exist in 3P? If that's the case, I feel it should be done.

Reply:
Yeah, `nextPageOperation` is a decorator in `Azure.Core`, and there aren't current plans to move it to TypeSpec core